animations/tween-animations.md

```diff
@@ -0,0 +1,727 @@
+but does not start it running:
+
+<!-- skip -->
+{% prettify dart %}
+final AnimationController controller = new AnimationController(
+    duration: const Duration(milliseconds: 2000), vsync: this);
+{% endprettify %}
+
+AnimationController derives from Animation<double>, so it can be used
+wherever an Animation object is needed. However, the AnimationController
+has additional methods to control the animation. For example, you start
+an animation with the `.forward()` method. The generation of numbers is
+tied to the screen refresh, so typically 60 numbers are generated per
+second. After each number is generated, each Animation object calls the
+attached Listener objects.
+
+When creating an AnimationController, you pass it a `vsync` argument.
+This option keeps the current widget tree in display memory until
+Flutter's rendering engine completes its refresh cycle. To use your
+custom State object as the `vsync`, include the `TickerProviderStateMixin`
+when defining the custom State class.
+
+## CurvedAnimation
+
+A CurvedAnimation modifies an Animation object's output to make the
+progression be a non-linear curve:
+
+<!-- skip -->
+{% prettify dart %}
+final CurvedAnimation curve =
+    new CurvedAnimation(parent: controller, curve: Curves.easeIn);
+{% endprettify %}
+
+CurvedAnimation and AnimationController are both of type
+Animation<double>, so you can pass them interchangeably.
+Note that the CurvedAnimation wraps the object it’s modifying.
+You don’t subclass AnimationController to implement a curve.
+
+<aside class="alert alert-success" markdown="1">
+By default, AnimationController can go from 0.0 to 1.0.
+There is also a `fling()` function that allows you to provide velocity,
+force, and position (via the Force object).
+The position can be anything and so can be outside of the 0.0 to 1.0 range.
+
+The CurvedAnimation can exceed its 0.0 to 1.0 range,
+even if the AnimationController doesn't. Depending on the curve selected,
+the output of the CurvedAnimation can have a wider range than the input.
+For example, elastic curves such as Curves.elasticIn will significantly
+overshoot or undershoot the default range.
+</aside>
+
+## Tween
+
+By default, the AnimationController object ranges from 0.0 to 1.0.
+If you need a different range or a different data type,
+you can use a Tween to configure an animation to interpolate to a
+different range or data type. For example, the following Tween
+goes from -200.0 to 0.0:
+
+<!-- skip -->
+{% prettify dart %}
+final Tween doubleTween = new Tween<double>(begin: -200.0, end: 0.0);
+{% endprettify %}
+
+A Tween is a stateless object that takes only `begin` and `end`.
+The sole job of a Tween is to define a mapping from an input range
+to an output range. The input range is commonly 0.0 to 1.0,
+but that’s not a requirement.
+
+A Tween inherits from Animatable<T>, not from Animation<T>.
+An Animatable does not have to output double.
```

Comment:
Nor does an `Animation<T>` (it produces a T). I'd probably move this sentence elsewhere lest people think that it is the explanation of the difference between Animation and Animatable.

Reply:
if you do want to explain the difference here, it's what you say in the paragraph below -- Animations have state and a current value and can be listened to, while Animatables describe how to map an `Animation<double>`'s value to another value.